docs/topics/arrays.md

```diff
@@ -31,32 +60,293 @@ fun main() {
+
+To return a pair of lists from an array of `Pair`, use the [unzip()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/unzip.html) function.
+
+```Kotlin
+fun main() { 
+//sampleStart
+    val pairArray = arrayOf(Pair(1,1), Pair(2,2), Pair(3,3))
+
+    // Unzips array ([1, 2, 3], [1, 2, 3])
+    val unzipArray = pairArray.unzip()
+    println(unzipArray)
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="unzip-array-kotlin"}
+
+## Concatenate arrays
+
+Arrays of type `String` can be concatenated using the `+` operator.
 
 ```kotlin
-val x: IntArray = intArrayOf(1, 2, 3)
-x[0] = x[1] + x[2]
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf("a", "b", "c")
+    val anotherArray = arrayOf("d", "e", "f")
+    
+    // Creates a new array by concatenating elements: ["a", "b", "c", "d", "e", "f"]
+    println((simpleArray + anotherArray).joinToString())
+//sampleEnd
+}
 ```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="concatenate-array-kotlin"}
+
+## Use arrays with variable number of arguments
+
+In some cases it is useful to be able to pass a variable number of arguments to a function without having to define the number of arguments in advance.
+In Kotlin you can use the `vararg` parameter for this. If you want to pass your variable number of arguments to a function as the contents of an array, use the _spread_ operator (`*`).
+The spread operator passes each element of your array as individual arguments to your chosen function.
+
+```kotlin
+fun main() {
+    val lettersArray = arrayOf("c", "d")
+    val lettersList = listOf("e", "f")
+    printAllStrings("a", "b", *lettersArray, *lettersList.toTypedArray())
+}
+
+fun printAllStrings(vararg strings: String) {
+    for(string in strings) {
+        println(string)
+    }
+}
+```
+
+For more information, see [Variable number of arguments (varargs)](functions.md#variable-number-of-arguments-varargs).
+
+## Convert to collection
+
+To convert an array to a [collection](collections-overview.md) (`List`, `Set` or `Map`), use the [`toList()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-list.html), [`toSet()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-set.html), [`toMap()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-map.html) functions respectively.
+
+> Only an array of [`Pair`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/) can be converted to a map: `Array<Pair<K,V>>`.
+> The first value of a `Pair` instance becomes a key and the second becomes a value.
+>
+{type="note"}
+
+```kotlin
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf("a", "b", "c", "c")
+    
+    // Convert to Set ["a", "b", "c"]
+    println(simpleArray.toSet())
+    
+    // Convert to List ["a", "b", "c", "c"]
+    println(simpleArray.toList())
+
+    val pairArray = arrayOf(Pair(1,1), Pair(2,2), Pair(3,3))
```

Comment:
Please use an example that is more expressive than just pairs of numbers. I am a fan of something like prizes `gold/silver/bronze -> 500/200/100`, or calories `apple/banana/kiwi -> 300/ 250/400`.
Bonus: add a second `kiwi` to your input array to illustrate that map keys are unique and will be overwritten.